x/nexus/keeper/transfer.go

```diff
@@ -84,27 +135,42 @@ func (k Keeper) EnqueueForTransfer(ctx sdk.Context, sender exported.CrossChainAd
 		return 0, fmt.Errorf("recipient's chain %s does not support foreign assets", recipient.Chain.Name)
 	}
 
+	// merging transfers below minimum for the specified recipient
+	insufficientAmountTransfer, found := k.getTransferForRecipientAndAsset(ctx, recipient, asset.Denom, exported.InsufficientAmount)
+	if found {
+		asset = asset.Add(insufficientAmountTransfer.Asset)
+		k.deleteTransfer(ctx, insufficientAmountTransfer)
+	}
+
 	// collect fee
-	if feeDue := sdk.NewDecFromInt(asset.Amount).Mul(feeRate).TruncateInt(); feeDue.IsPositive() {
-		k.addTransferFee(ctx, sdk.NewCoin(asset.Denom, feeDue))
-		asset = asset.SubAmount(feeDue)
+	fees, _ := k.ComputeTransferFee(ctx, sender.Chain, recipient.Chain, asset)
+	if fees.Amount.GTE(asset.Amount) {
+		k.Logger(ctx).Debug(fmt.Sprintf("skipping deposit for chain %s at %s from recipient %s due to deposited amount being below fees %s for asset %s",
+			sender.Chain.Name, sender.Address, recipient.Address, fees.String(), asset.String()))
+
+		return k.setNewTransfer(ctx, recipient, asset, exported.InsufficientAmount), nil
+	}
+
+	if fees.IsPositive() {
+		k.addTransferFee(ctx, fees)
+		asset = asset.Sub(fees)
 	}
 
 	// merging transfers for the specified recipient
-	previousTransfer, found := k.getPendingTransferForRecipientAndAsset(ctx, recipient, asset.Denom)
+	previousTransfer, found := k.getTransferForRecipientAndAsset(ctx, recipient, asset.Denom, exported.Pending)
 	if found {
 		asset = asset.Add(previousTransfer.Asset)
 		k.deleteTransfer(ctx, previousTransfer)
 	}
 
-	k.Logger(ctx).Info(fmt.Sprintf("Transfer of %s to cross chain address %s in %s successfully prepared",
-		asset.String(), recipient.Address, recipient.Chain.Name))
+	k.Logger(ctx).Info(fmt.Sprintf("Transfer of %s from %s in %s to cross chain address %s in %s successfully prepared",
+		asset.String(), sender.Address, sender.Chain.Name, recipient.Address, recipient.Chain.Name))
 
 	return k.setNewPendingTransfer(ctx, recipient, asset), nil
 }
 
-func (k Keeper) getPendingTransferForRecipientAndAsset(ctx sdk.Context, recipient exported.CrossChainAddress, denom string) (exported.CrossChainTransfer, bool) {
-	iter := k.getStore(ctx).Iterator(getTransferPrefix(recipient.Chain.Name, exported.Pending))
+func (k Keeper) getTransferForRecipientAndAsset(ctx sdk.Context, recipient exported.CrossChainAddress, denom string, state exported.TransferState) (exported.CrossChainTransfer, bool) {
```

Comment:
```suggestion
func (k Keeper) getTransfer(ctx sdk.Context, recipient exported.CrossChainAddress, denom string, state exported.TransferState) (exported.CrossChainTransfer, bool) {
```
the current name gives no helpful additional information